client/router.js

```diff
@@ -113,6 +113,8 @@ let router = new Router({
 router.beforeEach((to, from, next) => {
   if (to.matched.some(it => it.meta.auth) && !store.state.auth.user) {
     next({ path: '/login', query: { redirect: to.fullPath } });
+  } else if (to.matched.some(it => it.meta.restricted) && !store.getters.isAdmin) {
+    next({ name: from.name });
```

Comment:
Do we always have from? How about fallback to home if from isn't available for some obscure reason?